### ISSUE TITLE
Trying to understand max_overflow

### ISSUE THREAD

**Issue by wminghao** [devinus/poolboy#100]
Hi,

I am trying to understand max_overflow, 

1) What exactly does it do?
2) if size > max_overflow, is there any harm?
3) If poolsize too big, will it crash the erlang runtime?

Regards,
Thanks

**Comment by YongHaoWu** [comment]
@devinus  Am I right? I got this from reading code.

**Comment by zugolosian** [comment]
That sounds right ^ It'll only start overflow workers when all the original workers are checked out

**Comment by Vagabond** [comment]
Yes, overflow children are started on-demand (and destroyed on being checked back in).
